src/main/java/org/jenkinsci/plugins/workflow/steps/CoreWrapperStep.java

```diff
@@ -92,9 +93,26 @@ public SimpleBuildWrapper getDelegate() {
         }
 
         private void doStart() throws Exception {
-            SimpleBuildWrapper.Context c = new SimpleBuildWrapper.Context();
-            Run<?, ?> run = getContext().get(Run.class);
-            delegate.setUp(c, run, getContext().get(FilePath.class), getContext().get(Launcher.class), getContext().get(TaskListener.class), getContext().get(EnvVars.class));
+            // Get the context objects that may be required
+            final StepContext context = this.getContext();
+            final Run<?,?> run = context.get(Run.class);
+            final FilePath workspace = context.get(FilePath.class);
+            final Launcher launcher = context.get(Launcher.class);
+            final TaskListener listener = context.get(TaskListener.class);
+            final EnvVars env = context.get(EnvVars.class);
+            // Ensure those that are required are provided
+            Objects.requireNonNull(run);
```

Comment:
Fine but unnecessary: `getRequiredContext` handles this.

Reply:
Yes, but non in a way that is obvious from static analysis.

Reply:
Adjusted to use asserts instead.